// Copyright (C) 2020 Ellen Poe
//
// This file is part of MASS.
//
// MASS is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// MASS is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with MASS.  If not, see <http://www.gnu.org/licenses/>.

syntax = "proto3";
package api;

import "mass/api/chat.proto";

message DoActionRequest {
  // A repeated field containing any requests to the vessel's systems. May be
  // empty.
  repeated SystemRequest system_requests = 1;

  // Optional, a chat message to send to other players in the vessel.
  ChatMessage chat_message = 2;
}

message DoActionResponse {}

message SystemRequest {
  oneof request {
    // Steering request.
    SteeringSystemRequest steering_request = 1;
    // Diving request.
    DivingSystemRequest diving_request = 2;
    // Propulsion request.
    PropulsionSystemRequest propulsion_request = 3;
    // Map requests.
    MapSystemRequest map_request = 4;
  }
}

message SteeringSystemRequest {
  // Requested heading.
  uint32 heading_degrees = 1;
}

message DivingSystemRequest {
  // Requested depth, in feet.
  uint32 depth_feet = 1;
}

message PropulsionSystemRequest {
  // Requested speed, in knots.
  uint32 speed_knots = 1;
}

message MapSystemRequest {
  // Eventually put marking and removing POIs in here.
}
